### PR TITLE
Create a branch dedicated to builds of ArcticDB against `libevent 2.1.10`

### PR DESCRIPTION
**Pull request by jjerphan**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

**Context**

Builds of ArcticDB against libevent 2.1.10 are currently needed by some users of ArcticDB. [On `main`,there is s a build variant configuration to build ArcticDB against `libevent 2.1.10` and `libevent 2.1.12`](https://github.com/conda-forge/arcticdb-feedstock/blob/9021dcbc50c96eb8af498167d6085656299c8bf3/recipe/conda_build_config.yaml#L5-L7), but this causes several problems for conda-forge (see https://github.com/conda-forge/arcticdb-feedstock/pull/60 for an overview of them).

This PR (which aims at starting the maintenance of another branch named `libevent-2.1.10`) would unlock conda-forge's migration bot as mentioned by @h-vetinari in https://github.com/conda-forge/arcticdb-feedstock/pull/60#issuecomment-1728987525. Once it is merged, `libevent` can be unpinned from the configuration.

The channel will remain the default and ArcticDB will continue to be installable as it is actually the case with, that is with:

```
mamba install -c conda-forge arcticdb
```

This solution is courtesy of @jamesmunro, and was mentioned to be effective by @h-vetinari in discussions we had.

Since those builds are going to be uploaded on a dedicated channel, I am not updating the build number (it could be reset to 0), but I prefer to keep it in sync with the one on `main`. We will then need to update `main` and this new `libevent-2.1.10` branch synchronously.

@h-vetinari: can you explicitly approve this PR if this resolves the afore-referenced problem? Thank you.

### CODE REVIEW

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -4,6 +4,10 @@ MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
...
 libevent:
   - "2.1.10"
-  - "2.1.12"
+lmdb:
+  - "0.9.22"
 msgpack_c:
   - 6
+
+channel_targets:
+  - conda-forge arcticdb_libevent_2_1_10
```
Comment (h-vetinari) [inline recipe/conda_build_config.yaml:13]: You don't need to publish to a separate label, you can keep publishing to the main channel. My main concern was that the `main` branch stays up-to-date (more or less) with our migration, but what you do on non-`main` branches is less of an issue (and can still publish to the main channel/label regularly)

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (jjerphan) [reply recipe/conda_build_config.yaml]: Thank you for the information, I have not read the migration bot code and did not know that `main` was only the branch it considers.

**Pull request merged**